src/programs/RelationalPrograms.jl

```diff
@@ -145,9 +145,15 @@ function parse_relation_diagram(head::Expr, body::Expr)
     Expr(:where, expr, context) => (expr, parse_relation_context(context)...)
     _ => (head, nothing, nothing)
   end
-  var_types = if isnothing(all_types) # Untyped case.
+  var_types = if isnothing(all_types)    # Untyped case
     vars -> length(vars)
-  else # Typed case.
+  elseif typeof(all_types[1]) <: Int     # Int typed case
```

Comment:
you should use `eltype(all_types) <: Integer` here. Int is a concrete type so it only has itself as subtype. Int32 and Int64 are both subtypes of Integer. The `eltype` function get the element type of the array, which is more reliable than the type of the first element.